File: leetcode/217/contains-duplicate.ts
function containsDuplicate(nums: number[]): boolean {
  const map = new Map();
  for (const num of nums) {
    if (map.has(num)) {
      return true;
    }
    map.set(num, 1);
  }
  return false;
}

console.log(containsDuplicate([1, 2, 3, 1]));
console.log(containsDuplicate([1, 2, 3, 4]));
console.log(containsDuplicate([1, 1, 1, 3, 3, 4, 3, 2, 4, 2]));

File: leetcode/217/contains-duplicate.go
package main

func main() {
	
}

func containsDuplicate(nums []int) bool {
	numsMap := make(map[int]bool)

	for _, val := range nums {
		if _, exists := numsMap[val]; exists {
			return true
		}
		numsMap[val] = true
	}
	return false
}